python-api/tests/unittests/test_connection_unit.py

```diff
@@ -147,6 +147,59 @@ def test_mock2(self, monkeypatch):
         assert conn.connection is not None
 
 
+class TestSignalSubscribe(object):
+
+    """Unittests for signal_subscribe"""
+
+    def test_handler_not_callable(self, monkeypatch):
+        """Test if handler is not callable"""
+
+        conn = Connection()
+        test_cbs = ['', None, [], {}]
+        for test_cb in test_cbs:
+            with pytest.raises(ValueError):
+                conn.signal_subscribe(test_cb)
+
+    def test_handler_passed_to_gio(self, monkeypatch):
+        """Test if handler is correctly passed to Gio"""
+
+        monkeypatch.setattr(
+            Gio.DBusConnection, 'new_for_address_sync',
+            Mock(return_value=Gio.DBusConnection()))
+
+        SignalSubscribeMock = Mock()
+        monkeypatch.setattr(
+            Gio.DBusConnection, 'signal_subscribe',
+            SignalSubscribeMock)
+
+        conn = Connection()
+        conn.connect_dbus()
+        test_cb = lambda: None
```

Comment:
It doesn't matter because it is never called. The only thing verified here is, that the callback passed in is the same passed to Gio